tests/generators/random/generate.py

```diff
@@ -0,0 +1,245 @@
+"""
+This test format currently uses code generation to assemble the tests
+as the current test infra does not have a facility to dynamically
+generate tests that can be seen by ``pytest``.
+
+This will likley change in future releases of the testing infra.
+
+NOTE: To add additional scenarios, add test cases below in ``_generate_randomized_scenarios``.
+"""
+
+import sys
+import warnings
+from typing import Callable
+import itertools
+
+from eth2spec.test.utils.random import (
+    no_block,
+    no_op_validation,
+    randomize_state,
+    randomize_state_altair,
+    random_block,
+    random_block_altair,
+    last_slot_in_epoch,
+    random_slot_in_epoch,
+    penultimate_slot_in_epoch,
+    epoch_transition,
+    slot_transition,
+    transition_with_random_block,
+    transition_to_leaking,
+    transition_without_leak,
+)
+from eth2spec.test.helpers.constants import PHASE0, ALTAIR
+
+
+# Ensure this many blocks are present in *each* randomized scenario
```

Comment:
This is *precisely* this many blocks, rather than *at least* this many, right?

Variance here might be marginally valuable